lib/spack/spack/hooks/autopush.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import argparse
+
+import spack.mirror
+from spack.cmd import buildcache as bc
+
+
+def post_install_in_db(spec):
+    pkg = spec.package
+    # Package was installed from source
+    if not pkg.installed_from_binary_cache:
```

Comment:
nitpick about style:
```suggestion
    if pkg.installed_from_binary_cache:
        return
    # push
```